shared/constants/transaction.ts

```diff
@@ -0,0 +1,426 @@
+   * has an allowance of. For more information on allowances, see the approve
+   * type.
+   */
+  tokenMethodTransferFrom = 'transferfrom',
+  /**
+   * A token transaction requesting an allowance of all of a user's token to
+   * spend on behalf of the user
+   */
+  tokenMethodSetApprovalForAll = 'setapprovalforall',
+}
+
+/**
+ * In EIP-2718 typed transaction envelopes were specified, with the very first
+ * typed envelope being 'legacy' and describing the shape of the base
+ * transaction params that were hitherto the only transaction type sent on
+ * Ethereum.
+ */
+export enum TransactionEnvelopeType {
+  /**
+   * A legacy transaction, the very first type.
+   */
+  legacy = '0x0',
+  /**
+   * EIP-2930 defined the access list transaction type that allowed for
+   * specifying the state that a transaction would act upon in advance and
+   * theoretically save on gas fees.
+   */
+  accessList = '0x1',
+  /**
+   * The type introduced comes from EIP-1559, Fee Market describes the addition
+   * of a baseFee to blocks that will be burned instead of distributed to
+   * miners. Transactions of this type have both a maxFeePerGas (maximum total
+   * amount in gwei per gas to spend on the transaction) which is inclusive of
+   * the maxPriorityFeePerGas (maximum amount of gwei per gas from the
+   * transaction fee to distribute to miner).
+   */
+  feeMarket = '0x2',
+}
+
+/**
+ * Transaction Status is a mix of Ethereum and MetaMask terminology, used internally
+ * for transaction processing.
+ */
+export enum TransactionStatus {
+  /**
+   * A new transaction that the user has not approved or rejected
+   */
+  unapproved = 'unapproved',
+  /**
+   * The user has approved the transaction in the MetaMask UI
+   */
+  approved = 'approved',
+  /**
+   * The user has rejected the transaction in the MetaMask UI
+   */
+  rejected = 'rejected',
+  /**
+   * The transaction has been signed
+   */
+  signed = 'signed',
+  /**
+   * The transaction has been submitted to network
+   */
+  submitted = 'submitted',
+  /**
+   * The transaction has failed for some reason
+   */
+  failed = 'failed',
+  /**
+   * The transaction was dropped due to a tx with same nonce being accepted
+   */
+  dropped = 'dropped',
+  /**
+   * The transaction was confirmed by the network
+   */
+  confirmed = 'confirmed',
+  /**
+   * The transaction has been signed and is waiting to either be confirmed,
+   * dropped or failed. This is a "fake" status that we use to group statuses
+   * that are very similar from the user's perspective (approved,
+   * signed, submitted). The only notable case where approve and signed are
+   * different from user perspective is in hardware wallets where the
+   * transaction is signed on an external device. Otherwise signing happens
+   * transparently to users.
+   */
+  pending = 'pending',
+}
+
+/**
+ * With this list we can detect if a transaction is still in progress.
+ */
+export const IN_PROGRESS_TRANSACTION_STATUSES = [
```

Comment:
Question: should this be `InProgressTransactionStatuses` to be consistent with the rest type names?

Reply:
Thanks for  asking @DDDDDanica :) 

So the 'UpperCamelCase' convention is for TypeScript enum types only. The above is is a constant array so I kept it with our previous 'UPPER_SNAKE_CASE' convention.

Reply:
Makes sense. Thank for the explanation  🦾